packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -58,7 +64,83 @@ export async function fetchSnapshot(
     );
 }
 
-export async function fetchLatestSnapshotCore(
+export async function fetchSnapshotWithRedeem(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    snapshotOptions: ISnapshotOptions | undefined,
+    logger: ITelemetryLogger,
+    snapshotDownloader: (url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<IOdspSnapshot>>,
+    putInCache: (valueWithEpoch: IVersionedValueWithEpoch) => Promise<void>,
+    removeEntries: () => Promise<void>,
+): Promise<ISnapshotCacheValue> {
+    try {
+        try {
+            const odspSnapshot = await fetchLatestSnapshotCore(
+                odspResolvedUrl,
+                storageTokenFetcher,
+                snapshotOptions,
+                logger,
+                snapshotDownloader,
+                putInCache,
+            );
+            return odspSnapshot;
+        } catch(error) {
+            if (isRedeemSharingLinkError(odspResolvedUrl, error)) {
+                logger.sendErrorEvent({
+                    eventName: "RedeemFallback",
+                    errorType: error.errorType,
+                });
+                await redeemSharingLink(odspResolvedUrl, storageTokenFetcher, logger);
+                const odspResolvedUrlWithoutShareLink: IOdspResolvedUrl =
+                    { ...odspResolvedUrl, sharingLinkToRedeem: undefined };
+                const odspSnapshot = await fetchLatestSnapshotCore(
```

Comment:
you do not need extra variable here and above, just do
return await fetchLatestSnapshotCore(...)

Reply:
await with return gives error.